src/components/NotesCard.js

```diff
@@ -121,12 +143,60 @@ export const NotesCard = ({ noteDetails }) => {
             >
               label
             </i>
-            <i className="material-icons">archive</i>
             <i
               className="material-icons"
               onClick={() => {
-                toast.success("Deleted Note");
-                dispatchNotes({ type: "DELETE-NOTE", payload: _id });
+                if (isPinned && !isArchived) {
+                  dispatchNotes({ type: "ARCHIVE-NOTE", payload: _id });
+                  dispatchNotes({
+                    type: "NOTE-PIN",
+                    payload: {
+                      _id,
+                      title,
+                      body,
+                      isPinned,
+                      isArchived,
+                      isTrashed,
+                    },
+                  });
+
+                  toast.success("Note unpinned & Archived");
+                } else {
+                  toast.success(
+                    notesState.notesList[getIndexOfNote(_id)].isArchived
+                      ? `Note Unarchived: Moved to Home`
+                      : `Moved to Archive`
+                  );
+                  dispatchNotes({ type: "ARCHIVE-NOTE", payload: _id });
+                }
+              }}
+            >
+              {notesState.notesList[getIndexOfNote(_id)].isArchived
+                ? `unarchive`
+                : `archive`}
+            </i>
+            <i
+              className="material-icons"
+              onClick={() => {
+                if (isPinned && !isTrashed) {
+                  dispatchNotes({ type: "TRASH-NOTE", payload: _id });
+                  dispatchNotes({
+                    type: "NOTE-PIN",
+                    payload: {
+                      _id,
+                      title,
```

Comment:
You could have used a variable to pass object value to payload rather than writing each property, every time